{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:black;font-size:30pt\"> Mitochondria assembly and species identification\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\"> We start by assembling the raw nanopore reads. We use genomeSize=16k as it is the expected size of a mitochondrial genome. </span>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Canu snapshot v1.7 +373 changes (r9065 568e52e019308accfd9727269085fe64cb527581)\n",
      "--\n",
      "-- CITATIONS\n",
      "--\n",
      "-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.\n",
      "-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.\n",
      "-- Genome Res. 2017 May;27(5):722-736.\n",
      "-- http://doi.org/10.1101/gr.215087.116\n",
      "-- \n",
      "-- Read and contig alignments during correction, consensus and GFA building use:\n",
      "--   Šošic M, Šikic M.\n",
      "--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.\n",
      "--   Bioinformatics. 2017 May 1;33(9):1394-1395.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw753\n",
      "-- \n",
      "-- Overlaps are generated using:\n",
      "--   Berlin K, et al.\n",
      "--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.\n",
      "--   Nat Biotechnol. 2015 Jun;33(6):623-30.\n",
      "--   http://doi.org/10.1038/nbt.3238\n",
      "-- \n",
      "--   Myers EW, et al.\n",
      "--   A Whole-Genome Assembly of Drosophila.\n",
      "--   Science. 2000 Mar 24;287(5461):2196-204.\n",
      "--   http://doi.org/10.1126/science.287.5461.2196\n",
      "-- \n",
      "-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:\n",
      "--   Chin CS, et al.\n",
      "--   Phased diploid genome assembly with single-molecule real-time sequencing.\n",
      "--   Nat Methods. 2016 Dec;13(12):1050-1054.\n",
      "--   http://doi.org/10.1038/nmeth.4035\n",
      "-- \n",
      "-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:\n",
      "--   Chin CS, et al.\n",
      "--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.\n",
      "--   Nat Methods. 2013 Jun;10(6):563-9\n",
      "--   http://doi.org/10.1038/nmeth.2474\n",
      "-- \n",
      "-- CONFIGURE CANU\n",
      "--\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_121' (from 'java') with -d64 support.\n",
      "-- Detected gnuplot version '5.0 patchlevel 3' (from 'gnuplot') and image format 'png'.\n",
      "-- Detected 8 CPUs and 60 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "--                            (tag)Concurrency\n",
      "--                     (tag)Threads          |\n",
      "--            (tag)Memory         |          |\n",
      "--        (tag)         |         |          |     total usage     algorithm\n",
      "--        -------  ------  --------   --------  -----------------  -----------------------------\n",
      "-- Local: meryl      8 GB    4 CPUs x   1 job      8 GB    4 CPUs  (k-mer counting)\n",
      "-- Local: cormhap    6 GB    8 CPUs x   1 job      6 GB    8 CPUs  (overlap detection with mhap)\n",
      "-- Local: obtovl     4 GB    8 CPUs x   1 job      4 GB    8 CPUs  (overlap detection)\n",
      "-- Local: utgovl     4 GB    8 CPUs x   1 job      4 GB    8 CPUs  (overlap detection)\n",
      "-- Local: ovb        4 GB    1 CPU  x   8 jobs    32 GB    8 CPUs  (overlap store bucketizer)\n",
      "-- Local: ovs        8 GB    1 CPU  x   7 jobs    56 GB    7 CPUs  (overlap store sorting)\n",
      "-- Local: red        8 GB    4 CPUs x   2 jobs    16 GB    8 CPUs  (read error detection)\n",
      "-- Local: oea        4 GB    1 CPU  x   8 jobs    32 GB    8 CPUs  (overlap error adjustment)\n",
      "-- Local: bat       16 GB    4 CPUs x   1 job     16 GB    4 CPUs  (contig construction with bogart)\n",
      "-- Local: gfa        8 GB    4 CPUs x   1 job      8 GB    4 CPUs  (GFA alignment and processing)\n",
      "--\n",
      "-- Found Nanopore uncorrected reads in the input files.\n",
      "--\n",
      "-- Generating assembly 'sark' in '/home/ubuntu/git-presentations/shark_mito/assemble'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        16000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.3200 ( 32.00%)\n",
      "--    obtOvlErrorRate 0.1200 ( 12.00%)\n",
      "--    utgOvlErrorRate 0.1200 ( 12.00%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.5000 ( 50.00%)\n",
      "--    obtErrorRate    0.1200 ( 12.00%)\n",
      "--    utgErrorRate    0.1200 ( 12.00%)\n",
      "--    cnsErrorRate    0.2000 ( 20.00%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:08:18 2018 with 13.121 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/sqStoreCreate \\\n",
      "      -o ./sark.seqStore.BUILDING \\\n",
      "      -minlength 500 \\\n",
      "      ./sark.seqStore.ssi \\\n",
      "    > ./sark.seqStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:08:20 2018 (2 seconds) with 13.044 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In sequence store './sark.seqStore':\n",
      "--   Found 63728 reads.\n",
      "--   Found 228085909 bases (14255.36 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 848.64 reads):\n",
      "--        0   9999  59405 **********************************************************************\n",
      "--    10000  19999   3721 ****\n",
      "--    20000  29999    447 \n",
      "--    30000  39999     86 \n",
      "--    40000  49999     28 \n",
      "--    50000  59999     17 \n",
      "--    60000  69999      6 \n",
      "--    70000  79999      6 \n",
      "--    80000  89999      3 \n",
      "--    90000  99999      5 \n",
      "--   100000 109999      2 \n",
      "--   110000 119999      0 \n",
      "--   120000 129999      1 \n",
      "--   130000 139999      0 \n",
      "--   140000 149999      0 \n",
      "--   150000 159999      0 \n",
      "--   160000 169999      0 \n",
      "--   170000 179999      0 \n",
      "--   180000 189999      0 \n",
      "--   190000 199999      0 \n",
      "--   200000 209999      0 \n",
      "--   210000 219999      0 \n",
      "--   220000 229999      0 \n",
      "--   230000 239999      0 \n",
      "--   240000 249999      0 \n",
      "--   250000 259999      0 \n",
      "--   260000 269999      0 \n",
      "--   270000 279999      0 \n",
      "--   280000 289999      0 \n",
      "--   290000 299999      0 \n",
      "--   300000 309999      0 \n",
      "--   310000 319999      0 \n",
      "--   320000 329999      0 \n",
      "--   330000 339999      0 \n",
      "--   340000 349999      0 \n",
      "--   350000 359999      0 \n",
      "--   360000 369999      0 \n",
      "--   370000 379999      0 \n",
      "--   380000 389999      0 \n",
      "--   390000 399999      0 \n",
      "--   400000 409999      0 \n",
      "--   410000 419999      0 \n",
      "--   420000 429999      0 \n",
      "--   430000 439999      0 \n",
      "--   440000 449999      0 \n",
      "--   450000 459999      0 \n",
      "--   460000 469999      0 \n",
      "--   470000 479999      0 \n",
      "--   480000 489999      0 \n",
      "--   490000 499999      0 \n",
      "--   500000 509999      0 \n",
      "--   510000 519999      0 \n",
      "--   520000 529999      0 \n",
      "--   530000 539999      0 \n",
      "--   540000 549999      0 \n",
      "--   550000 559999      0 \n",
      "--   560000 569999      0 \n",
      "--   570000 579999      0 \n",
      "--   580000 589999      0 \n",
      "--   590000 599999      0 \n",
      "--   600000 609999      0 \n",
      "--   610000 619999      0 \n",
      "--   620000 629999      0 \n",
      "--   630000 639999      0 \n",
      "--   640000 649999      0 \n",
      "--   650000 659999      0 \n",
      "--   660000 669999      0 \n",
      "--   670000 679999      0 \n",
      "--   680000 689999      0 \n",
      "--   690000 699999      0 \n",
      "--   700000 709999      0 \n",
      "--   710000 719999      0 \n",
      "--   720000 729999      0 \n",
      "--   730000 739999      0 \n",
      "--   740000 749999      0 \n",
      "--   750000 759999      0 \n",
      "--   760000 769999      0 \n",
      "--   770000 779999      0 \n",
      "--   780000 789999      0 \n",
      "--   790000 799999      0 \n",
      "--   800000 809999      0 \n",
      "--   810000 819999      0 \n",
      "--   820000 829999      0 \n",
      "--   830000 839999      0 \n",
      "--   840000 849999      0 \n",
      "--   850000 859999      0 \n",
      "--   860000 869999      0 \n",
      "--   870000 879999      0 \n",
      "--   880000 889999      0 \n",
      "--   890000 899999      0 \n",
      "--   900000 909999      0 \n",
      "--   910000 919999      0 \n",
      "--   920000 929999      0 \n",
      "--   930000 939999      0 \n",
      "--   940000 949999      0 \n",
      "--   950000 959999      0 \n",
      "--   960000 969999      0 \n",
      "--   970000 979999      0 \n",
      "--   980000 989999      0 \n",
      "--   990000 999999      0 \n",
      "--   1000000 1009999      0 \n",
      "--   1010000 1019999      0 \n",
      "--   1020000 1029999      0 \n",
      "--   1030000 1039999      0 \n",
      "--   1040000 1049999      0 \n",
      "--   1050000 1059999      0 \n",
      "--   1060000 1069999      0 \n",
      "--   1070000 1079999      0 \n",
      "--   1080000 1089999      0 \n",
      "--   1090000 1099999      0 \n",
      "--   1100000 1109999      0 \n",
      "--   1110000 1119999      0 \n",
      "--   1120000 1129999      0 \n",
      "--   1130000 1139999      1 \n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Thu Aug 30 22:08:20 2018 with 13.043 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:08:58 2018 (38 seconds) with 12.345 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 154325019 *******************************************************************--> 0.8633 0.6795\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--       2-     2  18228584 ********************************************************************** 0.9652 0.8400\n",
      "--       3-     4   4594021 *****************                                                      0.9848 0.8862\n",
      "--       5-     7    937974 ***                                                                    0.9937 0.9163\n",
      "--       8-    11    321481 *                                                                      0.9969 0.9330\n",
      "--      12-    16    147175                                                                        0.9982 0.9440\n",
      "--      17-    22     76092                                                                        0.9989 0.9519\n",
      "--      23-    29     42673                                                                        0.9993 0.9579\n",
      "--      30-    37     26353                                                                        0.9995 0.9624\n",
      "--      38-    46     16902                                                                        0.9996 0.9661\n",
      "--      47-    56     12060                                                                        0.9997 0.9691\n",
      "--      57-    67      8246                                                                        0.9998 0.9717\n",
      "--      68-    79      6106                                                                        0.9998 0.9739\n",
      "--      80-    92      4565                                                                        0.9999 0.9759\n",
      "--      93-   106      3490                                                                        0.9999 0.9776\n",
      "--     107-   121      2633                                                                        0.9999 0.9791\n",
      "--     122-   137      2042                                                                        0.9999 0.9803\n",
      "--     138-   154      1698                                                                        0.9999 0.9815\n",
      "--     155-   172      1313                                                                        1.0000 0.9826\n",
      "--     173-   191      1060                                                                        1.0000 0.9835\n",
      "--     192-   211       916                                                                        1.0000 0.9844\n",
      "--     212-   232       689                                                                        1.0000 0.9852\n",
      "--     233-   254       597                                                                        1.0000 0.9858\n",
      "--     255-   277       472                                                                        1.0000 0.9865\n",
      "--     278-   301       401                                                                        1.0000 0.9870\n",
      "--     302-   326       348                                                                        1.0000 0.9875\n",
      "--     327-   352       286                                                                        1.0000 0.9880\n",
      "--     353-   379       243                                                                        1.0000 0.9884\n",
      "--     380-   407       214                                                                        1.0000 0.9888\n",
      "--     408-   436       191                                                                        1.0000 0.9892\n",
      "--     437-   466       166                                                                        1.0000 0.9895\n",
      "--     467-   497       164                                                                        1.0000 0.9899\n",
      "--     498-   529       148                                                                        1.0000 0.9902\n",
      "--     530-   562       113                                                                        1.0000 0.9905\n",
      "--     563-   596       113                                                                        1.0000 0.9908\n",
      "--     597-   631        94                                                                        1.0000 0.9911\n",
      "--     632-   667       100                                                                        1.0000 0.9914\n",
      "--     668-   704        79                                                                        1.0000 0.9916\n",
      "--     705-   742        78                                                                        1.0000 0.9919\n",
      "--     743-   781        60                                                                        1.0000 0.9921\n",
      "--     782-   821        60                                                                        1.0000 0.9923\n",
      "--\n",
      "--      148975 (max occurrences)\n",
      "--    72804970 (total mers, non-unique)\n",
      "--    24440731 (distinct mers, non-unique)\n",
      "--   154325019 (unique mers)\n",
      "-- For mhap overlapping, the threshold 0.000005000000000 is too low, resetting to 0.000156903543019 to capture 5.000000 * 14255 coverage.\n",
      "-- For mhap overlapping, set repeat k-mer threshold to 71275.\n",
      "--\n",
      "-- Found 227129989 16-mers; 178765750 distinct and 154325019 unique.  Largest count 148975.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=low based on read coverage of 14255.\n",
      "--\n",
      "-- PARAMETERS: hashes=256, minMatches=3, threshold=0.85\n",
      "--\n",
      "-- Given 5.4 GB, can fit 16200 reads per block.\n",
      "-- For 5 blocks, set stride to 2 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 4 mhap precompute jobs.\n",
      "-- Configured 5 mhap overlap jobs.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cormhap' concurrent execution on Thu Aug 30 22:09:02 2018 with 13.043 GB free disk space (4 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "    ./precompute.sh 3 > ./precompute.000003.out 2>&1\n",
      "    ./precompute.sh 4 > ./precompute.000004.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:16:49 2018 (467 seconds) with 12.037 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 4 mhap precompute jobs finished successfully.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cormhap' concurrent execution on Thu Aug 30 22:16:49 2018 with 12.037 GB free disk space (5 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "    ./mhap.sh 3 > ./mhap.000003.out 2>&1\n",
      "    ./mhap.sh 4 > ./mhap.000004.out 2>&1\n",
      "    ./mhap.sh 5 > ./mhap.000005.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:35 2018 (226 seconds) with 11.785 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 5 mhap overlap output files.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:20:35 2018 with 11.785 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreConfig \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -M 4-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     -create ./sark.ovlStore.config \\\n",
      "     > ./sark.ovlStore.config.txt \\\n",
      "    2> ./sark.ovlStore.config.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:35 2018 (like a bat out of hell) with 11.785 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Creating overlap store correction/sark.ovlStore using:\n",
      "--      2 buckets\n",
      "--      2 slices\n",
      "--        using at most 2 GB memory each\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovB' concurrent execution on Thu Aug 30 22:20:35 2018 with 11.785 GB free disk space (2 processes; 8 concurrently)\n",
      "\n",
      "    cd correction/sark.ovlStore.BUILDING\n",
      "    ./scripts/1-bucketize.sh 1 > ./logs/1-bucketize.000001.out 2>&1\n",
      "    ./scripts/1-bucketize.sh 2 > ./logs/1-bucketize.000002.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:40 2018 (5 seconds) with 11.296 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store bucketizer finished.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovS' concurrent execution on Thu Aug 30 22:20:40 2018 with 11.296 GB free disk space (2 processes; 7 concurrently)\n",
      "\n",
      "    cd correction/sark.ovlStore.BUILDING\n",
      "    ./scripts/2-sort.sh 1 > ./logs/2-sort.000001.out 2>&1\n",
      "    ./scripts/2-sort.sh 2 > ./logs/2-sort.000002.out 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Thu Aug 30 22:20:51 2018 (11 seconds) with 10.542 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store sorter finished.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:20:51 2018 with 10.542 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreIndexer \\\n",
      "      -O  ./sark.ovlStore.BUILDING \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -C  ./sark.ovlStore.config \\\n",
      "      -delete \\\n",
      "    > ./sark.ovlStore.BUILDING.index.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:51 2018 (lickety-split) with 11.053 GB free disk space\n",
      "----------------------------------------\n",
      "FROM: 'correction/sark.ovlStore.BUILDING'\n",
      "TO:   'correction/sark.ovlStore'\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:20:51 2018 with 11.053 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreDump \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -O  ./sark.ovlStore \\\n",
      "     -counts \\\n",
      "     > ./sark.ovlStore/counts.dat 2> ./sark.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:51 2018 (in the blink of an eye) with 11.053 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/sark.ovlStore' successfully constructed.\n",
      "-- Found 37104578 overlaps for 35361 reads; 28367 reads have no overlaps.\n",
      "--\n",
      "--\n",
      "-- Purged 2.74 GB in 20 overlap output files.\n",
      "-- Set corMinCoverage=4 based on read coverage of 14255.\n",
      "-- Global filter scores will be estimated.\n",
      "-- Computing correction layouts.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:20:52 2018 with 12.311 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/generateCorrectionLayouts \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -O  ./sark.ovlStore \\\n",
      "      -C  ./sark.corStore.WORKING \\\n",
      "      -eC 80 \\\n",
      "    > ./sark.corStore.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:58 2018 (6 seconds) with 12.266 GB free disk space\n",
      "----------------------------------------\n",
      "-- Computing correction layouts.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:20:58 2018 with 12.266 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/filterCorrectionLayouts \\\n",
      "      -S  ../../sark.seqStore \\\n",
      "      -C     ../sark.corStore \\\n",
      "      -R      ./sark.readsToCorrect.WORKING \\\n",
      "      -cc 4 \\\n",
      "      -cl 500 \\\n",
      "      -g  16000 \\\n",
      "      -c  40 \\\n",
      "    > ./sark.readsToCorrect.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:20:58 2018 (in the blink of an eye) with 12.262 GB free disk space\n",
      "----------------------------------------\n",
      "--                             original      original\n",
      "--                            raw reads     raw reads\n",
      "--   category                w/overlaps  w/o/overlaps\n",
      "--   -------------------- ------------- -------------\n",
      "--   Number of Reads              32906         30822\n",
      "--   Number of Bases           98557999       8222335\n",
      "--   Coverage                  6159.875       513.896\n",
      "--   Median                        1652             0\n",
      "--   Mean                          2995           266\n",
      "--   N50                           5268          4989\n",
      "--   Minimum                        500             0\n",
      "--   Maximum                     128708         33882\n",
      "--   \n",
      "--                                        --------corrected---------  ----------rescued----------\n",
      "--                             evidence                     expected                     expected\n",
      "--   category                     reads            raw     corrected            raw     corrected\n",
      "--   -------------------- -------------  ------------- -------------  ------------- -------------\n",
      "--   Number of Reads                197             18            18           3707          3707\n",
      "--   Number of Bases            2838998         887234        655682       18350676       8158531\n",
      "--   Coverage                   177.437         55.452        40.980       1146.917       509.908\n",
      "--   Median                        8054          37654         36546           3736          1056\n",
      "--   Mean                         14411          49290         36426           4950          2200\n",
      "--   N50                          26278          54878         37499           7385          4327\n",
      "--   Minimum                        823          26040         26027            503           501\n",
      "--   Maximum                     128708         128708         59378          31296         24457\n",
      "--   \n",
      "--                        --------uncorrected--------\n",
      "--                                           expected\n",
      "--   category                       raw     corrected\n",
      "--   -------------------- ------------- -------------\n",
      "--   Number of Reads              60003         60003\n",
      "--   Number of Bases           87542424       3220247\n",
      "--   Coverage                  5471.401       201.265\n",
      "--   Median                         557             0\n",
      "--   Mean                          1458            53\n",
      "--   N50                           4741           418\n",
      "--   Minimum                          0             0\n",
      "--   Maximum                     101843         73851\n",
      "--   \n",
      "--   Maximum Memory          1687255982\n",
      "--\n",
      "--                            (tag)Concurrency\n",
      "--                     (tag)Threads          |\n",
      "--            (tag)Memory         |          |\n",
      "--        (tag)         |         |          |     total usage     algorithm\n",
      "--        -------  ------  --------   --------  -----------------  -----------------------------\n",
      "-- Local: cor        6 GB    4 CPUs x   2 jobs    12 GB    8 CPUs  (read correction)\n",
      "--\n",
      "-- Using overlaps no worse than 0.5 fraction error for correcting reads (from corErrorRate parameter).\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cor' concurrent execution on Thu Aug 30 22:20:58 2018 with 12.262 GB free disk space (13 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "    ./correctReads.sh 3 > ./correctReads.000003.out 2>&1\n",
      "    ./correctReads.sh 4 > ./correctReads.000004.out 2>&1\n",
      "    ./correctReads.sh 5 > ./correctReads.000005.out 2>&1\n",
      "    ./correctReads.sh 6 > ./correctReads.000006.out 2>&1\n",
      "    ./correctReads.sh 7 > ./correctReads.000007.out 2>&1\n",
      "    ./correctReads.sh 8 > ./correctReads.000008.out 2>&1\n",
      "    ./correctReads.sh 9 > ./correctReads.000009.out 2>&1\n",
      "    ./correctReads.sh 10 > ./correctReads.000010.out 2>&1\n",
      "    ./correctReads.sh 11 > ./correctReads.000011.out 2>&1\n",
      "    ./correctReads.sh 12 > ./correctReads.000012.out 2>&1\n",
      "    ./correctReads.sh 13 > ./correctReads.000013.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:43 2018 (105 seconds) with 12.24 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 13 read correction output files.\n",
      "-- Found 13 read correction output files.\n",
      "--\n",
      "-- Loading corrected reads into corStore and seqStore.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:22:43 2018 with 12.24 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/loadCorrectedReads \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -C ./sark.corStore \\\n",
      "      -L ./2-correction/corjob.files \\\n",
      "    >  ./sark.loadCorrectedReads.log \\\n",
      "    2> ./sark.loadCorrectedReads.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:43 2018 (in the blink of an eye) with 12.231 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In sequence store './sark.seqStore':\n",
      "--   Found 3281 reads.\n",
      "--   Found 7210813 bases (450.67 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 22.01 reads):\n",
      "--        0    999   1541 **********************************************************************\n",
      "--     1000   1999    821 *************************************\n",
      "--     2000   2999    298 *************\n",
      "--     3000   3999    153 ******\n",
      "--     4000   4999    130 *****\n",
      "--     5000   5999     70 ***\n",
      "--     6000   6999     61 **\n",
      "--     7000   7999     42 *\n",
      "--     8000   8999     35 *\n",
      "--     9000   9999     24 *\n",
      "--    10000  10999     28 *\n",
      "--    11000  11999     13 \n",
      "--    12000  12999     12 \n",
      "--    13000  13999      8 \n",
      "--    14000  14999      7 \n",
      "--    15000  15999      7 \n",
      "--    16000  16999      9 \n",
      "--    17000  17999      2 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--    18000  18999      3 \n",
      "--    19000  19999      1 \n",
      "--    20000  20999      0 \n",
      "--    21000  21999      1 \n",
      "--    22000  22999      1 \n",
      "--    23000  23999      3 \n",
      "--    24000  24999      2 \n",
      "--    25000  25999      1 \n",
      "--    26000  26999      1 \n",
      "--    27000  27999      1 \n",
      "--    28000  28999      1 \n",
      "--    29000  29999      1 \n",
      "--    30000  30999      1 \n",
      "--    31000  31999      1 \n",
      "--    32000  32999      0 \n",
      "--    33000  33999      1 \n",
      "--    34000  34999      0 \n",
      "--    35000  35999      0 \n",
      "--    36000  36999      0 \n",
      "--    37000  37999      0 \n",
      "--    38000  38999      0 \n",
      "--    39000  39999      0 \n",
      "--    40000  40999      0 \n",
      "--    41000  41999      0 \n",
      "--    42000  42999      0 \n",
      "--    43000  43999      0 \n",
      "--    44000  44999      0 \n",
      "--    45000  45999      0 \n",
      "--    46000  46999      0 \n",
      "--    47000  47999      0 \n",
      "--    48000  48999      0 \n",
      "--    49000  49999      0 \n",
      "--    50000  50999      0 \n",
      "--    51000  51999      1 \n",
      "--\n",
      "-- Purging correctReads output after loading into stores.\n",
      "-- Purged 13 .cns outputs.\n",
      "-- Purged 26 .out job log outputs.\n",
      "--\n",
      "-- Purging overlaps used for correction.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:22:44 2018 with 12.984 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/sqStoreDumpFASTQ \\\n",
      "      -corrected \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -o ./sark.correctedReads.gz \\\n",
      "      -fasta \\\n",
      "      -nolibname \\\n",
      "    > sark.correctedReads.fasta.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:44 2018 (fast as lightning) with 12.982 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Corrected reads saved in 'sark.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Thu Aug 30 22:22:44 2018 with 12.982 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:44 2018 (fast as lightning) with 12.963 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   3018816 *******************************************************************--> 0.8621 0.4227\n",
      "--       2-     2    217062 ********************************************************************** 0.9241 0.4835\n",
      "--       3-     4    119394 **************************************                                 0.9465 0.5164\n",
      "--       5-     7     56169 ******************                                                     0.9657 0.5576\n",
      "--       8-    11     30151 *********                                                              0.9771 0.5958\n",
      "--      12-    16     17335 *****                                                                  0.9841 0.6311\n",
      "--      17-    22     11322 ***                                                                    0.9885 0.6629\n",
      "--      23-    29     10593 ***                                                                    0.9915 0.6925\n",
      "--      30-    37      9515 ***                                                                    0.9946 0.7332\n",
      "--      38-    46      2786                                                                        0.9969 0.7723\n",
      "--      47-    56      1704                                                                        0.9976 0.7872\n",
      "--      57-    67      1284                                                                        0.9981 0.7993\n",
      "--      68-    79       985                                                                        0.9985 0.8100\n",
      "--      80-    92       751                                                                        0.9987 0.8199\n",
      "--      93-   106       549                                                                        0.9989 0.8287\n",
      "--     107-   121       441                                                                        0.9991 0.8362\n",
      "--     122-   137       372                                                                        0.9992 0.8433\n",
      "--     138-   154       229                                                                        0.9993 0.8498\n",
      "--     155-   172       259                                                                        0.9994 0.8545\n",
      "--     173-   191       189                                                                        0.9995 0.8602\n",
      "--     192-   211       215                                                                        0.9995 0.8651\n",
      "--     212-   232       160                                                                        0.9996 0.8711\n",
      "--     233-   254       123                                                                        0.9996 0.8761\n",
      "--     255-   277       142                                                                        0.9997 0.8803\n",
      "--     278-   301       108                                                                        0.9997 0.8855\n",
      "--     302-   326       106                                                                        0.9997 0.8898\n",
      "--     327-   352        91                                                                        0.9998 0.8946\n",
      "--     353-   379        57                                                                        0.9998 0.8988\n",
      "--     380-   407        65                                                                        0.9998 0.9018\n",
      "--     408-   436        61                                                                        0.9998 0.9053\n",
      "--     437-   466        62                                                                        0.9998 0.9091\n",
      "--     467-   497        57                                                                        0.9999 0.9128\n",
      "--     498-   529        40                                                                        0.9999 0.9167\n",
      "--     530-   562        65                                                                        0.9999 0.9196\n",
      "--     563-   596        41                                                                        0.9999 0.9245\n",
      "--     597-   631        18                                                                        0.9999 0.9278\n",
      "--     632-   667        20                                                                        0.9999 0.9294\n",
      "--     668-   704        26                                                                        0.9999 0.9313\n",
      "--     705-   742        20                                                                        0.9999 0.9337\n",
      "--     743-   781        20                                                                        0.9999 0.9357\n",
      "--     782-   821        15                                                                        0.9999 0.9379\n",
      "--\n",
      "--       51885 (max occurrences)\n",
      "--     4123107 (total mers, non-unique)\n",
      "--      482779 (distinct mers, non-unique)\n",
      "--     3018816 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:22:45 2018 with 12.963 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/meryl-san \\\n",
      "      -Dt \\\n",
      "      -n 1377 \\\n",
      "      -s ./sark.ms22 \\\n",
      "    > ./sark.ms22.frequentMers.fasta \\\n",
      "    2> ./sark.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:45 2018 (in the blink of an eye) with 12.963 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 1377.\n",
      "--\n",
      "-- Found 7141923 22-mers; 3501595 distinct and 3018816 unique.  Largest count 51885.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.12\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:22:45 2018 with 12.982 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/overlapInCorePartition \\\n",
      "     -S  ../../sark.seqStore \\\n",
      "     -hl 85312000 \\\n",
      "     -rl 1000000000 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./sark.partition \\\n",
      "    > ./sark.partition.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:22:45 2018 (like a bat out of hell) with 12.982 GB free disk space\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'obtovl' concurrent execution on Thu Aug 30 22:22:45 2018 with 12.982 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:45 2018 (480 seconds) with 12.98 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           548287      548287 +- 0\n",
      "--     with an overlap            98431       98431 +- 0\n",
      "--\n",
      "--   overlaps                     98431       98431 +- 0\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:45 2018 with 12.98 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreConfig \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -M 2 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     -create ./sark.ovlStore.config \\\n",
      "     > ./sark.ovlStore.config.txt \\\n",
      "    2> ./sark.ovlStore.config.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:45 2018 (like a bat out of hell) with 12.98 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Creating overlap store trimming/sark.ovlStore using:\n",
      "--      1 bucket\n",
      "--      2 slices\n",
      "--        using at most 1 GB memory each\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovB' concurrent execution on Thu Aug 30 22:30:45 2018 with 12.98 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd trimming/sark.ovlStore.BUILDING\n",
      "    ./scripts/1-bucketize.sh 1 > ./logs/1-bucketize.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:45 2018 (like a bat out of hell) with 12.977 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store bucketizer finished.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovS' concurrent execution on Thu Aug 30 22:30:45 2018 with 12.977 GB free disk space (2 processes; 7 concurrently)\n",
      "\n",
      "    cd trimming/sark.ovlStore.BUILDING\n",
      "    ./scripts/2-sort.sh 1 > ./logs/2-sort.000001.out 2>&1\n",
      "    ./scripts/2-sort.sh 2 > ./logs/2-sort.000002.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:45 2018 (in the blink of an eye) with 12.965 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store sorter finished.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:45 2018 with 12.965 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreIndexer \\\n",
      "      -O  ./sark.ovlStore.BUILDING \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -C  ./sark.ovlStore.config \\\n",
      "      -delete \\\n",
      "    > ./sark.ovlStore.BUILDING.index.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:45 2018 (fast as lightning) with 12.97 GB free disk space\n",
      "----------------------------------------\n",
      "FROM: 'trimming/sark.ovlStore.BUILDING'\n",
      "TO:   'trimming/sark.ovlStore'\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:46 2018 with 12.97 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreDump \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -O  ./sark.ovlStore \\\n",
      "     -counts \\\n",
      "     > ./sark.ovlStore/counts.dat 2> ./sark.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:46 2018 (one second) with 12.97 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/sark.ovlStore' successfully constructed.\n",
      "-- Found 196862 overlaps for 1700 reads; 62028 reads have no overlaps.\n",
      "--\n",
      "--\n",
      "-- Purged 0.001 GB in 3 overlap output files.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:46 2018 with 12.972 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/trimReads \\\n",
      "      -S  ../../sark.seqStore \\\n",
      "      -O  ../sark.ovlStore \\\n",
      "      -Co ./sark.1.trimReads.clear \\\n",
      "      -e  0.12 \\\n",
      "      -minlength 500 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./sark.1.trimReads \\\n",
      "    >     ./sark.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:47 2018 (one second) with 12.97 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--      500    (reads trimmed below this many bases are deleted)\n",
      "--   0.1200    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   63728 reads      7210813 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--    1536 reads      2158742 bases (trimmed reads output)\n",
      "--     164 reads       219071 bases (reads with no change, kept as is)\n",
      "--   62028 reads      4306901 bases (reads with no overlaps, deleted)\n",
      "--       0 reads            0 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--    1227 reads       274298 bases (bases trimmed from the 5' end of a read)\n",
      "--    1187 reads       251801 bases (bases trimmed from the 3' end of a read)\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:47 2018 with 12.97 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/splitReads \\\n",
      "      -S  ../../sark.seqStore \\\n",
      "      -O  ../sark.ovlStore \\\n",
      "      -Ci ./sark.1.trimReads.clear \\\n",
      "      -Co ./sark.2.splitReads.clear \\\n",
      "      -e  0.12 \\\n",
      "      -minlength 500 \\\n",
      "      -o  ./sark.2.splitReads \\\n",
      "    >     ./sark.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:47 2018 (lickety-split) with 12.97 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--      500    (reads trimmed below this many bases are deleted)\n",
      "--   0.1200    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--    1700 reads      2903912 bases (reads processed)\n",
      "--   62028 reads      4306901 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--       4 reads         4557 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--    1696 reads      2899355 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--       0 reads            0 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--       0 reads            0 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--       0 reads            0 bases (trimmed from the 5' end of the read)\n",
      "--       0 reads            0 bases (trimmed from the 3' end of the read)\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:47 2018 with 12.97 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/loadTrimmedReads \\\n",
      "      -S ../../sark.seqStore \\\n",
      "      -c ./sark.2.splitReads.clear \\\n",
      "    > ./sark.loadtrimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:47 2018 (lickety-split) with 12.97 GB free disk space\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "--\n",
      "-- In sequence store './sark.seqStore':\n",
      "--   Found 1700 reads.\n",
      "--   Found 2377813 bases (148.61 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 15.07 reads):\n",
      "--        0    999   1055 **********************************************************************\n",
      "--     1000   1999    459 ******************************\n",
      "--     2000   2999     84 *****\n",
      "--     3000   3999     22 *\n",
      "--     4000   4999     13 \n",
      "--     5000   5999     11 \n",
      "--     6000   6999      9 \n",
      "--     7000   7999      6 \n",
      "--     8000   8999      7 \n",
      "--     9000   9999      1 \n",
      "--    10000  10999      7 \n",
      "--    11000  11999      2 \n",
      "--    12000  12999      4 \n",
      "--    13000  13999      4 \n",
      "--    14000  14999      3 \n",
      "--    15000  15999      4 \n",
      "--    16000  16999      9 \n",
      "--\n",
      "-- Purging overlaps used for trimming.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:47 2018 with 12.979 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/sqStoreDumpFASTQ \\\n",
      "      -trimmed \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -o ./sark.trimmedReads.gz \\\n",
      "      -fasta \\\n",
      "      -nolibname \\\n",
      "    > ./sark.trimmedReads.fasta.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:47 2018 (in the blink of an eye) with 12.979 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Trimmed reads saved in 'sark.trimmedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Thu Aug 30 22:30:47 2018 with 12.979 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:48 2018 (one second) with 12.974 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1    486224 *******************************************************************--> 0.7306 0.2076\n",
      "--       2-     2     64134 ********************************************************************** 0.8270 0.2624\n",
      "--       3-     4     41887 *********************************************                          0.8676 0.2969\n",
      "--       5-     7     22742 ************************                                               0.9056 0.3446\n",
      "--       8-    11     14484 ***************                                                        0.9312 0.3945\n",
      "--      12-    16      9341 **********                                                             0.9493 0.4471\n",
      "--      17-    22      6664 *******                                                                0.9622 0.5015\n",
      "--      23-    29      7521 ********                                                               0.9716 0.5561\n",
      "--      30-    37      7418 ********                                                               0.9833 0.6464\n",
      "--      38-    46      1438 *                                                                      0.9929 0.7379\n",
      "--      47-    56       753                                                                        0.9947 0.7601\n",
      "--      57-    67       539                                                                        0.9958 0.7756\n",
      "--      68-    79       416                                                                        0.9966 0.7901\n",
      "--      80-    92       383                                                                        0.9972 0.8036\n",
      "--      93-   106       248                                                                        0.9977 0.8167\n",
      "--     107-   121       178                                                                        0.9981 0.8269\n",
      "--     122-   137       102                                                                        0.9984 0.8353\n",
      "--     138-   154        93                                                                        0.9985 0.8407\n",
      "--     155-   172        67                                                                        0.9986 0.8465\n",
      "--     173-   191       102                                                                        0.9988 0.8515\n",
      "--     192-   211        71                                                                        0.9989 0.8592\n",
      "--     212-   232        89                                                                        0.9990 0.8650\n",
      "--     233-   254        67                                                                        0.9991 0.8737\n",
      "--     255-   277        53                                                                        0.9992 0.8805\n",
      "--     278-   301        54                                                                        0.9993 0.8865\n",
      "--     302-   326        61                                                                        0.9994 0.8937\n",
      "--     327-   352        27                                                                        0.9995 0.9012\n",
      "--     353-   379        33                                                                        0.9995 0.9056\n",
      "--     380-   407        66                                                                        0.9996 0.9106\n",
      "--     408-   436        25                                                                        0.9997 0.9214\n",
      "--     437-   466        17                                                                        0.9997 0.9261\n",
      "--     467-   497        23                                                                        0.9997 0.9291\n",
      "--     498-   529        21                                                                        0.9998 0.9343\n",
      "--     530-   562        14                                                                        0.9998 0.9387\n",
      "--     563-   596         5                                                                        0.9998 0.9417\n",
      "--     597-   631         7                                                                        0.9998 0.9430\n",
      "--     632-   667         2                                                                        0.9998 0.9449\n",
      "--     668-   704         6                                                                        0.9999 0.9454\n",
      "--     705-   742        12                                                                        0.9999 0.9472\n",
      "--     743-   781        10                                                                        0.9999 0.9509\n",
      "--     782-   821         7                                                                        0.9999 0.9542\n",
      "--\n",
      "--        7962 (max occurrences)\n",
      "--     1855889 (total mers, non-unique)\n",
      "--      179244 (distinct mers, non-unique)\n",
      "--      486224 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:48 2018 with 12.974 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/meryl-san \\\n",
      "      -Dt \\\n",
      "      -n 1112 \\\n",
      "      -s ./sark.ms22 \\\n",
      "    > ./sark.ms22.frequentMers.fasta \\\n",
      "    2> ./sark.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:48 2018 (lickety-split) with 12.974 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 1112.\n",
      "--\n",
      "-- Found 2342113 22-mers; 665468 distinct and 486224 unique.  Largest count 7962.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.12\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:30:48 2018 with 12.979 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/overlapInCorePartition \\\n",
      "     -S  ../../sark.seqStore \\\n",
      "     -hl 85312000 \\\n",
      "     -rl 1000000000 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./sark.partition \\\n",
      "    > ./sark.partition.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:30:48 2018 (in the blink of an eye) with 12.979 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'utgovl' concurrent execution on Thu Aug 30 22:30:48 2018 with 12.979 GB free disk space (1 processes; 1 concurrently)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:47 2018 (59 seconds) with 12.976 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           114204      114204 +- 0\n",
      "--     with an overlap           151058      151058 +- 0\n",
      "--\n",
      "--   overlaps                    151058      151058 +- 0\n",
      "--     contained                  79552       79552 +- 0\n",
      "--     dovetail                   71506       71506 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:31:47 2018 with 12.976 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreConfig \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -M 1 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     -create ./sark.ovlStore.config \\\n",
      "     > ./sark.ovlStore.config.txt \\\n",
      "    2> ./sark.ovlStore.config.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:47 2018 (fuirously fast) with 12.976 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Creating overlap store unitigging/sark.ovlStore using:\n",
      "--      1 bucket\n",
      "--      2 slices\n",
      "--        using at most 1 GB memory each\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovB' concurrent execution on Thu Aug 30 22:31:47 2018 with 12.976 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/sark.ovlStore.BUILDING\n",
      "    ./scripts/1-bucketize.sh 1 > ./logs/1-bucketize.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:48 2018 (one second) with 12.971 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store bucketizer finished.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'ovS' concurrent execution on Thu Aug 30 22:31:48 2018 with 12.971 GB free disk space (2 processes; 7 concurrently)\n",
      "\n",
      "    cd unitigging/sark.ovlStore.BUILDING\n",
      "    ./scripts/2-sort.sh 1 > ./logs/2-sort.000001.out 2>&1\n",
      "    ./scripts/2-sort.sh 2 > ./logs/2-sort.000002.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:48 2018 (like a bat out of hell) with 12.958 GB free disk space\n",
      "----------------------------------------\n",
      "-- Overlap store sorter finished.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:31:48 2018 with 12.958 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreIndexer \\\n",
      "      -O  ./sark.ovlStore.BUILDING \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -C  ./sark.ovlStore.config \\\n",
      "      -delete \\\n",
      "    > ./sark.ovlStore.BUILDING.index.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:48 2018 (in the blink of an eye) with 12.965 GB free disk space\n",
      "----------------------------------------\n",
      "FROM: 'unitigging/sark.ovlStore.BUILDING'\n",
      "TO:   'unitigging/sark.ovlStore'\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:31:48 2018 with 12.965 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreDump \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -O  ./sark.ovlStore \\\n",
      "     -counts \\\n",
      "     > ./sark.ovlStore/counts.dat 2> ./sark.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:48 2018 (fast as lightning) with 12.964 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/sark.ovlStore' successfully constructed.\n",
      "-- Found 302116 overlaps for 1697 reads; 62031 reads have no overlaps.\n",
      "--\n",
      "--\n",
      "-- Purged 0.002 GB in 3 overlap output files.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:31:48 2018 with 12.967 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/ovStoreStats \\\n",
      "     -C 148 \\\n",
      "     -S ../sark.seqStore \\\n",
      "     -O  ./sark.ovlStore \\\n",
      "     -o  ./sark.ovlStore \\\n",
      "     > ./sark.ovlStore.summary.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:48 2018 (fuirously fast) with 12.967 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/sark.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing          8    0.47     2916.38 +- 1823.25        250.12 +- 285.25     (bad trimming)\n",
      "--   middle-hump             4    0.24     1038.25 +- 612.53           0.25 +- 0.50       (bad trimming)\n",
      "--   no-5-prime             17    1.00     1773.53 +- 1234.29        123.35 +- 185.58     (bad trimming)\n",
      "--   no-3-prime             16    0.94     1912.62 +- 1709.90         49.50 +- 82.61      (bad trimming)\n",
      "--   \n",
      "--   low-coverage          588   34.59     2120.04 +- 3254.35         22.10 +- 10.98      (easy to assemble, potential for lower quality consensus)\n",
      "--   unique                210   12.35      719.81 +- 197.41         121.56 +- 47.34      (easy to assemble, perfect, yay)\n",
      "--   repeat-cont            95    5.59      631.95 +- 86.33          407.25 +- 67.87      (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat           232   13.65     1189.86 +- 556.09         744.06 +- 354.64     (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont      521   30.65     1038.02 +- 466.19                              (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove        6    0.35     2173.17 +- 678.72                              (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (repeat read, with unique section, probable bad read)\n",
      "--\n",
      "-- Loading read lengths.\n",
      "-- Loading number of overlaps per read.\n",
      "--\n",
      "-- Configure RED for 8gb memory.\n",
      "--                   Batches of at most (unlimited) reads.\n",
      "--                                      500000000 bases.\n",
      "--                   Expecting evidence of at most 536870912 bases per iteration.\n",
      "--\n",
      "--           Total                                               Reads                 Olaps Evidence\n",
      "--    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)\n",
      "--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------\n",
      "--      1  3102.72         1-63728          1700      2377813    27.27       302116     3.46  1024.00\n",
      "--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------\n",
      "--                                                    2377813                302116\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'red' concurrent execution on Thu Aug 30 22:31:49 2018 with 12.967 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:31:54 2018 (5 seconds) with 12.966 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 read error detection output files.\n",
      "--\n",
      "-- Loading read lengths.\n",
      "-- Loading number of overlaps per read.\n",
      "--\n",
      "-- Configure OEA for 4gb memory.\n",
      "--                   Batches of at most (unlimited) reads.\n",
      "--                                      300000000 bases.\n",
      "--\n",
      "--           Total                                               Reads                 Olaps  Adjusts\n",
      "--    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)\n",
      "--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------\n",
      "--      1  2389.11         1-63728          1700      2377813    10.32       302116     9.22   321.57\n",
      "--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------\n",
      "--                                                    2377813                302116\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'oea' concurrent execution on Thu Aug 30 22:31:55 2018 with 12.966 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:08 2018 (13 seconds) with 12.966 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlap error adjustment output files.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:08 2018 with 12.966 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/loadErates \\\n",
      "      -S ../../sark.seqStore \\\n",
      "      -O ../sark.ovlStore \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:08 2018 (like a bat out of hell) with 12.965 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'bat' concurrent execution on Thu Aug 30 22:32:08 2018 with 12.965 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:09 2018 (one second) with 12.963 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      17 sequences, total length 82177 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  432 sequences, total length 689319 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size 16kbp:\n",
      "--\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10       26786             1       26786\n",
      "--     20       26786             1       26786\n",
      "--     30       26786             1       26786\n",
      "--     40       26786             1       26786\n",
      "--     50       26786             1       26786\n",
      "--     60       26786             1       26786\n",
      "--     70       26786             1       26786\n",
      "--     80       26786             1       26786\n",
      "--     90       26786             1       26786\n",
      "--    100       26786             1       26786\n",
      "--    110       26786             1       26786\n",
      "--    120       26786             1       26786\n",
      "--    130       26786             1       26786\n",
      "--    140       26786             1       26786\n",
      "--    150       26786             1       26786\n",
      "--    160       26786             1       26786\n",
      "--    170       22247             2       49033\n",
      "--    180       22247             2       49033\n",
      "--    190       22247             2       49033\n",
      "--    200       22247             2       49033\n",
      "--    210       22247             2       49033\n",
      "--    220       22247             2       49033\n",
      "--    230       22247             2       49033\n",
      "--    240       22247             2       49033\n",
      "--    250       22247             2       49033\n",
      "--    260       22247             2       49033\n",
      "--    270       22247             2       49033\n",
      "--    280       22247             2       49033\n",
      "--    290       22247             2       49033\n",
      "--    300       22247             2       49033\n",
      "--    310        5266             3       54299\n",
      "--    320        5266             3       54299\n",
      "--    330        5266             3       54299\n",
      "--    340        4424             4       58723\n",
      "--    350        4424             4       58723\n",
      "--    360        4424             4       58723\n",
      "--    370        2860             5       61583\n",
      "--    380        2860             5       61583\n",
      "--    390        2781             6       64364\n",
      "--    400        2781             6       64364\n",
      "--    410        2610             7       66974\n",
      "--    420        2537             8       69511\n",
      "--    430        2537             8       69511\n",
      "--    440        2343             9       71854\n",
      "--    450        2208            10       74062\n",
      "--    460        2208            10       74062\n",
      "--    470        1791            11       75853\n",
      "--    480        1626            12       77479\n",
      "--    490        1588            13       79067\n",
      "--    500        1247            14       80314\n",
      "--    510         651            16       81617\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:09 2018 with 12.963 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/sqStoreCreatePartition \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T  ./sark.ctgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./sark.ctgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:09 2018 (fuirously fast) with 12.959 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:09 2018 with 12.959 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/sqStoreCreatePartition \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T  ./sark.utgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./sark.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:09 2018 (lickety-split) with 12.956 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '0').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "--\n",
      "--                            (tag)Concurrency\n",
      "--                     (tag)Threads          |\n",
      "--            (tag)Memory         |          |\n",
      "--        (tag)         |         |          |     total usage     algorithm\n",
      "--        -------  ------  --------   --------  -----------------  -----------------------------\n",
      "-- Local: cns        1 GB    4 CPUs x   2 jobs     2 GB    8 CPUs  (consensus)\n",
      "--\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cns' concurrent execution on Thu Aug 30 22:32:09 2018 with 12.956 GB free disk space (2 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (2 seconds) with 12.954 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "-- All 2 consensus jobs finished successfully.\n",
      "-- Using slow alignment for consensus (iteration '0').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.954 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreLoad \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T  ./sark.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (fast as lightning) with 12.953 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.953 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreLoad \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T  ./sark.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (fast as lightning) with 12.952 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "-- Purged 2 .cns outputs.\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.961 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T ./sark.ctgStore 2 \\\n",
      "      -sizes -s 16000 \\\n",
      "    > ./sark.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (like a bat out of hell) with 12.961 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      17 sequences, total length 82497 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  432 sequences, total length 689188 bp.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--\n",
      "-- Contig sizes based on genome size 16kbp:\n",
      "--\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10       26939             1       26939\n",
      "--     20       26939             1       26939\n",
      "--     30       26939             1       26939\n",
      "--     40       26939             1       26939\n",
      "--     50       26939             1       26939\n",
      "--     60       26939             1       26939\n",
      "--     70       26939             1       26939\n",
      "--     80       26939             1       26939\n",
      "--     90       26939             1       26939\n",
      "--    100       26939             1       26939\n",
      "--    110       26939             1       26939\n",
      "--    120       26939             1       26939\n",
      "--    130       26939             1       26939\n",
      "--    140       26939             1       26939\n",
      "--    150       26939             1       26939\n",
      "--    160       26939             1       26939\n",
      "--    170       22380             2       49319\n",
      "--    180       22380             2       49319\n",
      "--    190       22380             2       49319\n",
      "--    200       22380             2       49319\n",
      "--    210       22380             2       49319\n",
      "--    220       22380             2       49319\n",
      "--    230       22380             2       49319\n",
      "--    240       22380             2       49319\n",
      "--    250       22380             2       49319\n",
      "--    260       22380             2       49319\n",
      "--    270       22380             2       49319\n",
      "--    280       22380             2       49319\n",
      "--    290       22380             2       49319\n",
      "--    300       22380             2       49319\n",
      "--    310        5656             3       54975\n",
      "--    320        5656             3       54975\n",
      "--    330        5656             3       54975\n",
      "--    340        5656             3       54975\n",
      "--    350        4401             4       59376\n",
      "--    360        4401             4       59376\n",
      "--    370        4401             4       59376\n",
      "--    380        2855             5       62231\n",
      "--    390        2757             6       64988\n",
      "--    400        2757             6       64988\n",
      "--    410        2534             7       67522\n",
      "--    420        2534             7       67522\n",
      "--    430        2505             8       70027\n",
      "--    440        2346             9       72373\n",
      "--    450        2346             9       72373\n",
      "--    460        2188            10       74561\n",
      "--    470        1773            11       76334\n",
      "--    480        1608            12       77942\n",
      "--    490        1578            13       79520\n",
      "--    500        1125            14       80645\n",
      "--    510         644            16       81940\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.961 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreCoverageStat \\\n",
      "      -S ../sark.seqStore \\\n",
      "      -T  ./sark.ctgStore 2 \\\n",
      "      -s 16000 \\\n",
      "      -o ./sark.ctgStore.coverageStat \\\n",
      "    > ./sark.ctgStore.coverageStat.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (in the blink of an eye) with 12.961 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.961 GB free disk space\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh \\\n",
      "    > alignGFA.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (fast as lightning) with 12.961 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.961 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -T ./unitigging/sark.ctgStore 2 \\\n",
      "      -o ./sark.contigs \\\n",
      "      -layout \\\n",
      "    > ./sark.contigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (like a bat out of hell) with 12.959 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.959 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -T ./unitigging/sark.utgStore 2 \\\n",
      "      -o ./sark.unitigs \\\n",
      "      -layout \\\n",
      "    > ./sark.unitigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (in the blink of an eye) with 12.959 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.959 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -T ./unitigging/sark.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./sark.unassembled.fasta\n",
      "    2> ./sark.unassembled.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (fuirously fast) with 12.958 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.958 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -T ./unitigging/sark.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./sark.contigs.fasta\n",
      "    2> ./sark.contigs.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (in the blink of an eye) with 12.958 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Thu Aug 30 22:32:11 2018 with 12.958 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /home/ubuntu/canu/Linux-amd64/bin/tgStoreDump \\\n",
      "      -S ./sark.seqStore \\\n",
      "      -T ./unitigging/sark.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./sark.unitigs.fasta\n",
      "    2> ./sark.unitigs.err\n",
      "\n",
      "-- Finished on Thu Aug 30 22:32:11 2018 (lickety-split) with 12.958 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Assembly 'sark' finished in '/home/ubuntu/git-presentations/shark_mito/assemble'.\n",
      "--\n",
      "-- Summary saved in 'sark.report'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> 'sark.contigs.fasta'\n",
      "--   Unassembled   -> 'sark.unassembled.fasta'\n",
      "--   Unitigs       -> 'sark.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> 'sark.contigs.layout'.\n",
      "--   Unitigs       -> 'sark.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> 'sark.contigs.gfa'.\n",
      "--   Unitigs       -> 'sark.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n"
     ]
    }
   ],
   "source": [
    " ~/canu/Linux-amd64/bin/canu -d assemble/ -p shark genomeSize=16k minReadLength=500 -nanopore-raw all_shark.fastq\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">we then blast the contigs agains nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "export BLASTDB=/home/ubuntu/DB\n",
    "blastn -db nt -query shark.contigs.fasta -out contigs_nt.blastn -outfmt \"6 qseqid sseqid pident length evalue salltitles sallacc sstart send qstart qend\" -num_threads 16 -culling_limit 1 -evalue 0.001\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000001\tgi|19067985|gb|AY049832.1|\t93.27\t4025\t0.0\tGaleocerdo cuvier internal transcribed spacer 2, and 28S ribosomal RNA gene, partial sequence\tAY049832\t3987\t9\t258\t4191\n",
      "tig00000001\tgi|19067986|gb|AY049833.1|\t92.22\t1787\t0.0\tGaleocerdo cuvier 18S ribosomal RNA gene, complete sequence\tAY049833\t1780\t1\t9695\t11431\n",
      "tig00000001\tgi|501418415|gb|KC686343.1|\t86.01\t2358\t0.0\tOncorhynchus mykiss clone 450D22 zinc finger CCHC domain-containing protein 3 pseudogene, complete sequence\tKC686343\t9561\t11890\t1628\t3915\n",
      "tig00000001\tgi|501418415|gb|KC686343.1|\t88.33\t1354\t0.0\tOncorhynchus mykiss clone 450D22 zinc finger CCHC domain-containing protein 3 pseudogene, complete sequence\tKC686343\t7951\t9293\t214\t1523\n",
      "tig00000001\tgi|1220469783|gb|CP020797.1|\t89.89\t1602\t0.0\tOryzias latipes strain HNI chromosome 19\tCP020797\t954244\t955840\t9671\t11218\n",
      "tig00000001\tgi|22128507|gb|AF513986.1|\t92.54\t1300\t0.0\tCarcharhinus falciformis 5.8S ribosomal RNA gene, partial sequence; internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence\tAF513986\t1291\t1\t3894\t5141\n",
      "tig00000001\tgi|840013939|emb|LK065000.1|\t85.88\t793\t0.0\tApteryx australis mantelli genome assembly AptMant0, scaffold scaffold418\tLK065000\t66965\t67746\t10687\t11454\n",
      "tig00000001\tgi|1032033216|gb|KX121612.1|\t88.91\t487\t1e-162\tMus musculus clone contig 3 chromocenter region genomic sequence\tKX121612\t136\t620\t3433\t3909\n",
      "tig00000001\tgi|1032033216|gb|KX121612.1|\t91.35\t185\t7e-60\tMus musculus clone contig 3 chromocenter region genomic sequence\tKX121612\t1702\t1885\t5129\t5308\n",
      "tig00000001\tgi|338784686|gb|JN008719.1|\t78.61\t991\t4e-152\tRhizoprionodon porosus isolate OC5 5.8S ribosomal RNA gene, partial sequence; internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence\tJN008719\t959\t1\t4365\t5278\n",
      "tig00000001\tgi|685044968|emb|LN593493.1|\t85.29\t571\t2e-150\tCyprinus carpio genome assembly common carp genome ,scaffold 000001641\tLN593493\t62005\t62540\t452\t1014\n",
      "tig00000001\tgi|165974169|dbj|AB375548.1|\t79.61\t358\t6e-56\tTriakis scyllium genes for 18S rRNA, ITS1, 5.8S rRNA, partial and complete sequence\tAB375548\t1242\t892\t7145\t7485\n",
      "tig00000001\tgi|165974169|dbj|AB375548.1|\t87.30\t63\t2e-06\tTriakis scyllium genes for 18S rRNA, ITS1, 5.8S rRNA, partial and complete sequence\tAB375548\t184\t123\t9508\t9570\n",
      "tig00000001\tgi|1220469770|gb|CP020784.1|\t76.00\t350\t6e-31\tOryzias latipes strain HNI chromosome 6\tCP020784\t21720364\t21720702\t225\t554\n",
      "tig00000001\tgi|1167602264|gb|KY462798.1|\t89.53\t86\t5e-17\tAscotremella faginea voucher GM-2015-10-28 small subunit ribosomal RNA gene, partial sequence; internal transcribed spacer 1, 5.8S ribosomal RNA gene, and internal transcribed spacer 2, complete sequence; and large subunit ribosomal RNA gene, partial sequence\tKY462798\t96\t12\t9645\t9727\n",
      "tig00000001\tgi|1059793942|gb|AH007407.2|\t88.37\t86\t8e-15\tXenopus laevis x.laevis rrna non-transcribed spacer sequence, clone pxl108\tAH007407\t212\t127\t214\t294\n",
      "tig00000001\tgi|690413466|emb|LM976901.1|\t91.30\t69\t1e-12\tProtopolystoma xenopodis genome assembly P_xenopodis_South_Africa ,scaffold PXEA_contig0228254\tLM976901\t45\t111\t1554\t1620\n",
      "tig00000001\tgi|667700208|gb|KJ522988.1|\t92.06\t63\t2e-11\tAnolis humilis isolate 38017 18S ribosomal RNA gene, partial sequence; internal transcribed spacer 1, complete sequence; and 5.8S ribosomal RNA gene, partial sequence\tKJ522988\t1193\t1132\t5248\t5310\n",
      "tig00000001\tgi|110277094|gb|DQ660358.1|\t97.44\t39\t2e-05\tDicentrarchus labrax clone 436 28S ribosomal RNA gene, partial sequence; 3' external transcribed spacer, intergenic spacer, and 5' external transcribed spacer, complete sequence; and 18S ribosomal RNA gene, partial sequence\tDQ660358\t2243\t2205\t11416\t11454\n",
      "tig00000003\tgi|156119673|gb|AC164927.3|\t72.46\t973\t4e-68\tGinglymostoma cirratum clone GC_Ba-557B6, complete sequence\tAC164927\t38782\t39736\t307\t1253\n",
      "tig00000003\tgi|301128892|emb|FQ032660.1|\t96.15\t52\t2e-11\tScyliorhinus canicula Cluster_HOXD sequence\tFQ032660\t24936\t24987\t3\t53\n",
      "tig00000003\tgi|20197405|gb|AC005561.3|\t91.30\t46\t9e-05\tArabidopsis thaliana chromosome 2 clone T9F8 map mi310, complete sequence\tAC005561\t32479\t32435\t1\t45\n",
      "tig00000010\tgi|456368036|gb|KC470543.1|\t93.08\t14116\t0.0\tCarcharhinus obscurus mitochondrion, complete genome\tKC470543\t1\t14102\t8310\t22174\n",
      "tig00000010\tgi|456368036|gb|KC470543.1|\t91.96\t8474\t0.0\tCarcharhinus obscurus mitochondrion, complete genome\tKC470543\t8239\t16706\t1\t8309\n",
      "tig00000010\tgi|1382859157|gb|MG912791.1|\t73.40\t7698\t0.0\tChelonoidis duncanensis isolate 8291 mitochondrion, partial genome\tMG912791\t14022\t6492\t10735\t18206\n",
      "tig00000010\tgi|545484876|gb|KC992998.1|\t77.87\t2910\t0.0\tPoecilia mexicana limantouri mitochondrion, partial genome\tKC992998\t11334\t8442\t13352\t16210\n",
      "tig00000010\tgi|1034703429|gb|KX129904.1|\t73.45\t4626\t0.0\tChanna micropeltes mitochondrion, complete genome\tKX129904\t5069\t541\t8381\t12875\n",
      "tig00000010\tgi|1034703429|gb|KX129904.1|\t73.65\t1533\t5e-145\tChanna micropeltes mitochondrion, complete genome\tKX129904\t7721\t6213\t5632\t7121\n",
      "tig00000010\tgi|1276851520|gb|MG241667.1|\t93.63\t942\t0.0\tCarcharhinus galapagensis voucher CB7 D-loop, partial sequence; mitochondrial\tMG241667\t940\t1\t7323\t8238\n",
      "tig00000010\tgi|1018242234|dbj|LC137015.1|\t75.11\t1816\t0.0\tSalmo trutta fario mitochondrial DNA, complete genome, collection_date: 2014-11-19\tLC137015\t14427\t12645\t8790\t10545\n",
      "tig00000010\tgi|359300992|gb|JN315440.1|\t90.02\t591\t0.0\tSphyrna lewini isolate 107 cytochrome c oxidase subunit I gene, partial cds; mitochondrial\tJN315440\t591\t1\t13757\t14343\n",
      "tig00000010\tgi|1000233674|gb|KT696221.1|\t75.65\t1084\t6e-135\tHydrophilidae sp. BMNH 1274335 mitochondrion, partial genome\tKT696221\t13792\t12727\t14081\t15137\n",
      "tig00000010\tgi|1182510757|gb|KY117585.1|\t73.61\t1133\t2e-104\tSundamys muelleri mitochondrion, complete genome\tKY117585\t3162\t2046\t5926\t7031\n",
      "tig00000010\tgi|1269216117|gb|MF573068.1|\t73.67\t1128\t3e-102\tCapra hircus mitochondrion, complete genome\tMF573068\t8394\t7289\t245\t1340\n",
      "tig00000010\tgi|1353793188|gb|CP027098.1|\t74.97\t831\t1e-91\tBos mutus isolate yakQH1 chromosome X\tCP027098\t108172975\t108172155\t20311\t21119\n",
      "tig00000010\tgi|955487831|ref|XR_001389697.1|\t75.25\t804\t6e-90\tPREDICTED: Ceratotherium simum simum uncharacterized LOC106800237 (LOC106800237), ncRNA\tXR_001389697\t825\t41\t3919\t4704\n",
      "tig00000010\tgi|1033989031|gb|KX061088.1|\t76.09\t686\t2e-84\tPseudolyciella stylata isolate MO013e cytochrome oxidase subunit I gene, partial cds; mitochondrial\tKX061088\t801\t130\t14435\t15102\n",
      "tig00000010\tgi|27463608|gb|AF508069.1|\t82.28\t378\t5e-81\tPuntius sophore 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product\tAF508069\t415\t39\t8806\t9175\n",
      "tig00000010\tgi|850496270|gb|CP011912.1|\t71.65\t984\t5e-76\tOvis canadensis canadensis isolate 43U chromosome X sequence\tCP011912\t50685306\t50684337\t3860\t4817\n",
      "tig00000010\tgi|1182510771|gb|KY117583.1|\t71.70\t1279\t3e-68\tSundamys infraluteus mitochondrion, partial genome\tKY117583\t3441\t2187\t16581\t17795\n",
      "tig00000010\tgi|438879|gb|L24776.1|RATTPMGA\t84.54\t207\t4e-47\tRattus norvegicus tropomyosin non-muscle isoform NM3 (TPM-gamma) mRNA, complete cds\tL24776\t1060\t854\t11656\t11862\n",
      "tig00000010\tgi|31505609|gb|AY288049.1|\t79.57\t186\t6e-25\tBufo woodhousii isolate 1TR cytochrome b (cytb) gene, partial cds; mitochondrial gene for mitochondrial product\tAY288049\t188\t4\t6763\t6947\n",
      "tig00000017\tgi|858971314|gb|KP128093.1|\t81.98\t860\t0.0\tChiloscyllium punctatum antithrombin (serpinc1) gene, complete cds\tKP128093\t3920\t3078\t299\t1130\n",
      "tig00000017\tgi|858971314|gb|KP128093.1|\t81.18\t797\t9e-167\tChiloscyllium punctatum antithrombin (serpinc1) gene, complete cds\tKP128093\t3920\t3148\t645\t1419\n",
      "tig00000017\tgi|858971314|gb|KP128093.1|\t84.82\t606\t3e-162\tChiloscyllium punctatum antithrombin (serpinc1) gene, complete cds\tKP128093\t3679\t3078\t1\t583\n",
      "tig00000019\tgi|1016246|gb|U34992.1|CPU34992\t90.21\t562\t0.0\tCarcharhinus plumbeus Ig lambda light chain gene, complete cds\tU34992\t6638\t6080\t1355\t1892\n",
      "tig00000019\tgi|41469002|gb|AC144758.2|\t88.68\t53\t3e-04\tDanio rerio, complete sequence\tAC144758\t30664\t30714\t1637\t1686\n",
      "tig00000026\tgi|115334359|dbj|AB254130.1|\t81.19\t787\t2e-166\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t4433\t5216\t15\t766\n",
      "tig00000026\tgi|301128892|emb|FQ032660.1|\t94.00\t50\t1e-08\tScyliorhinus canicula Cluster_HOXD sequence\tFQ032660\t24937\t24986\t2\t50\n",
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t88.04\t276\t5e-84\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t794\t1067\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t87.10\t279\t5e-79\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t43\t315\n",
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t86.40\t272\t5e-74\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t647\t915\n",
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t86.30\t270\t2e-72\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t498\t760\n",
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t84.59\t279\t3e-66\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t193\t461\n",
      "tig00000756\tgi|115334359|dbj|AB254130.1|\t88.59\t184\t2e-52\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5797\t999\t1177\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t205\t263\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t652\t710\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t754\t812\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.28\t58\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t806\t863\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t857\t915\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t909\t967\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.28\t58\t3e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1112\t1169\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t98.28\t58\t9e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t5\t61\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t104\t162\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t257\t315\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t357\t414\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t6e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t553\t610\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t2e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t503\t559\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t2e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t704\t760\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t2e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1011\t1067\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t7e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t455\t509\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t7e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t604\t658\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t93.22\t59\t9e-12\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t65153\t65211\t55\t110\n",
      "tig00000756\tgi|301128880|emb|FQ032659.1|\t93.10\t58\t3e-11\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t309\t363\n",
      "tig00000756\tgi|1379072074|gb|MH015179.1|\t90.91\t77\t3e-16\tScyliorhinus canicula Fam45a protein (Fam45a), Eif3a protein (Eif3a), and Nanos1A protein (Nanos1A) genes, complete cds\tMH015179\t43947\t43871\t438\t509\n",
      "tig00000756\tgi|1379072074|gb|MH015179.1|\t93.65\t63\t2e-14\tScyliorhinus canicula Fam45a protein (Fam45a), Eif3a protein (Eif3a), and Nanos1A protein (Nanos1A) genes, complete cds\tMH015179\t43933\t43871\t1\t61\n",
      "tig00000756\tgi|1379072074|gb|MH015179.1|\t89.58\t48\t2e-04\tScyliorhinus canicula Fam45a protein (Fam45a), Eif3a protein (Eif3a), and Nanos1A protein (Nanos1A) genes, complete cds\tMH015179\t16460\t16507\t1071\t1117\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t87.36\t277\t2e-79\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t566\t835\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t87.87\t272\t2e-79\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t920\t1184\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t86.02\t279\t3e-73\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t314\t584\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t86.40\t272\t3e-72\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t1071\t1333\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t87.30\t244\t7e-69\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5730\t1371\t1609\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t84.64\t280\t1e-66\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t111\t382\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t84.23\t279\t8e-64\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t767\t1033\n",
      "tig00000757\tgi|115334359|dbj|AB254130.1|\t83.87\t279\t3e-62\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5706\t1217\t1483\n",
      "tig00000757\tgi|350626506|gb|JN646104.1|\t80.75\t161\t8e-24\tTriaenodon obesus selenium binding protein 1 (SBP1) mRNA, complete cds\tJN646104\t2226\t2066\t1\t159\n",
      "tig00000757\tgi|350626506|gb|JN646104.1|\t95.00\t40\t2e-05\tTriaenodon obesus selenium binding protein 1 (SBP1) mRNA, complete cds\tJN646104\t2222\t2183\t1617\t1656\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t100.00\t58\t8e-19\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t728\t785\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t100.00\t58\t8e-19\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1178\t1235\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t100.00\t57\t3e-18\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78982\t79038\t1376\t1432\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t1e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t65153\t65211\t71\t129\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t1e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t65153\t65211\t478\t536\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t1e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t65153\t65211\t975\t1033\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t98.31\t59\t1e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t65153\t65211\t1076\t1134\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t100.00\t55\t4e-17\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79035\t376\t430\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t98.28\t58\t1e-16\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t276\t332\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t2e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t627\t684\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t2e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1477\t1534\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t6e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t326\t382\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t6e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t876\t931\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t6e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t925\t981\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t8e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t19\t77\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t8e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t427\t484\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t8e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1426\t1483\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t3e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t578\t633\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t3e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t779\t835\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t3e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1027\t1082\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t94.83\t58\t3e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1229\t1284\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t93.22\t59\t4e-12\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t173\t231\n",
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t87.93\t58\t5e-06\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t530\t584\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000757\tgi|301128880|emb|FQ032659.1|\t86.44\t59\t2e-04\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t829\t882\n",
      "tig00000757\tgi|1379072074|gb|MH015179.1|\t96.61\t59\t2e-15\tScyliorhinus canicula Fam45a protein (Fam45a), Eif3a protein (Eif3a), and Nanos1A protein (Nanos1A) genes, complete cds\tMH015179\t1749\t1807\t123\t179\n",
      "tig00000758\tgi|115334359|dbj|AB254130.1|\t86.40\t272\t2e-73\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t255\t521\n",
      "tig00000758\tgi|115334359|dbj|AB254130.1|\t86.03\t272\t1e-71\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5973\t5706\t104\t370\n",
      "tig00000758\tgi|115334359|dbj|AB254130.1|\t83.64\t275\t5e-60\tSphyrna lewini GRLN gene for preproghrelin, complete cds\tAB254130\t5980\t5710\t801\t1064\n",
      "tig00000758\tgi|30578123|dbj|AB074142.1|\t90.61\t213\t5e-70\tTriakis scyllium IL-1 gene for interleukin-1beta, complete cds\tAB074142\t3476\t3266\t1069\t1277\n",
      "tig00000758\tgi|350626506|gb|JN646104.1|\t84.57\t162\t7e-34\tTriaenodon obesus selenium binding protein 1 (SBP1) mRNA, complete cds\tJN646104\t2228\t2067\t642\t799\n",
      "tig00000758\tgi|350626506|gb|JN646104.1|\t85.71\t147\t3e-32\tTriaenodon obesus selenium binding protein 1 (SBP1) mRNA, complete cds\tJN646104\t2212\t2066\t4\t147\n",
      "tig00000758\tgi|350626506|gb|JN646104.1|\t92.75\t69\t4e-16\tTriaenodon obesus selenium binding protein 1 (SBP1) mRNA, complete cds\tJN646104\t2229\t2162\t1346\t1414\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.67\t60\t4e-16\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1218\t1277\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t8\t66\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t159\t217\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t260\t318\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t312\t370\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.55\t58\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t413\t470\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t96.61\t59\t1e-15\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t1117\t1175\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t2e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78980\t79038\t862\t920\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t95.00\t60\t2e-14\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t962\t1021\n",
      "tig00000758\tgi|301128880|emb|FQ032659.1|\t94.92\t59\t2e-13\tScyliorhinus canicula Cluster_HOXB sequence\tFQ032659\t78981\t79038\t109\t165\n",
      "tig00000758\tgi|1160132575|ref|XR_002258344.1|\t88.89\t63\t2e-09\tPREDICTED: Rhincodon typus uncharacterized LOC109913723 (LOC109913723), ncRNA\tXR_002258344\t1\t62\t1361\t1423\n",
      "tig00000761\tgi|27549546|gb|AY172838.1|\t91.29\t1102\t0.0\tCarcharhinus plumbeus recombination activating gene 1 (RAG1) gene, partial cds; and recombination activating gene 2 (RAG2) gene, complete cds\tAY172838\t8301\t7202\t1150\t2200\n",
      "tig00000761\tgi|339895689|dbj|AB571627.1|\t77.46\t417\t2e-56\tGinglymostoma cirratum ZFP112, b2m, brd2 genes for similar to zinc finger protein 112, beta 2 microglobulin, bromodomain containing 2, complete cds\tAB571627\t74638\t75051\t1793\t2194\n",
      "tig00000762\tgi|27549546|gb|AY172838.1|\t93.86\t1091\t0.0\tCarcharhinus plumbeus recombination activating gene 1 (RAG1) gene, partial cds; and recombination activating gene 2 (RAG2) gene, complete cds\tAY172838\t8297\t7207\t499\t1563\n",
      "tig00000762\tgi|339895689|dbj|AB571627.1|\t78.47\t418\t1e-63\tGinglymostoma cirratum ZFP112, b2m, brd2 genes for similar to zinc finger protein 112, beta 2 microglobulin, bromodomain containing 2, complete cds\tAB571627\t74638\t75051\t1159\t1562\n",
      "tig00000762\tgi|262360117|gb|AC238881.1|\t89.09\t55\t7e-07\tCallorhinchus milii BAC clone IMCB_Eshark-67H12 from chromosome unknown, complete sequence\tAC238881\t102151\t102205\t1305\t1359\n"
     ]
    }
   ],
   "source": [
    "cat contigs_nt.blastn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\"> tig00000010 has a long hit to \"Carcharhinus obscurus mitochondrion\". let's extract it </span>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools faidx shark.contigs.fasta 'tig00000010' > tig00000010.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\"> And let's also download the Carcharhinus obscurus mitochondrion genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "curl 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=KC470543&retnode=text&rettype=fasta' > 'Carcharhinus_obscurus_mitochondrion.fasta' 2> /dev/null\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">Using Gepard we construct a dotplot of tig00000010 against itself. Note that the start and end of the sequence is highly similar. This is common when you assemble a circular genome \n",
    "\n",
    "<img src=\"files/tig10dotplot.jpeg\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">We then align the contig agains itself to find the ends of the repeated regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1: PREPARING DATA\n",
      "2,3: RUNNING mummer AND CREATING CLUSTERS\n",
      "# reading input file \"out.ntref\" of length 22175\n",
      "# construct suffix tree for sequence of length 22175\n",
      "# (maximum reference length is 536870908)\n",
      "# (maximum query length is 4294967295)\n",
      "# CONSTRUCTIONTIME /usr/bin/mummer out.ntref 0.00\n",
      "# reading input file \"/home/ubuntu/git-presentations/shark_mito/tig00000010.fasta\" of length 22174\n",
      "# matching query-file \"/home/ubuntu/git-presentations/shark_mito/tig00000010.fasta\"\n",
      "# against subject-file \"out.ntref\"\n",
      "# COMPLETETIME /usr/bin/mummer out.ntref 0.01\n",
      "# SPACE /usr/bin/mummer out.ntref 0.04\n",
      "4: FINISHING DATA\n",
      "1\t22174\t1\t22174\t22174\t22174\t100.00\t22174\t22174\t100.00\t100.00\ttig00000010\ttig00000010\n",
      "1\t5753\t16477\t22174\t5753\t5698\t94.58\t22174\t22174\t25.94\t25.70\ttig00000010\ttig00000010\n",
      "16477\t22174\t1\t5753\t5698\t5753\t94.58\t22174\t22174\t25.70\t25.94\ttig00000010\ttig00000010\n"
     ]
    }
   ],
   "source": [
    "nucmer --maxmatch --nosimplify tig00000010.fasta tig00000010.fasta\n",
    "show-coords -lrcTH out.delta\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">This shows that region 1-5753 is 94.58% similar to region 16477-22174. We use samtools to delete region 16477-22174"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools faidx tig00000010.fasta tig00000010:1-16476 > tig00000010_cut.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">The dotplot show no repeated regeions\n",
    "<img src=\"files/tig10_cut.jpeg\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">Aligning agains the Carcharhinus obscurus mitocondrion genome shows that tig00000010 is shifted. By convention, mitochondrial genomes are reported as lineal genomes starting on the tRNA-phe gene. Canu has no way of finding this gene, so the starting point is mostly random. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "<img src=\"files/tig10_vs_obscurus.jpeg\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">We aling Carcharhinus obscurus mitochondrion genome to tig00000010 to find the breaking point. In this case, it is between positions 8309 and 8310"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1: PREPARING DATA\n",
      "2,3: RUNNING mummer AND CREATING CLUSTERS\n",
      "# reading input file \"cut_vs_carcharhinus.ntref\" of length 16477\n",
      "# construct suffix tree for sequence of length 16477\n",
      "# (maximum reference length is 536870908)\n",
      "# (maximum query length is 4294967295)\n",
      "# CONSTRUCTIONTIME /usr/bin/mummer cut_vs_carcharhinus.ntref 0.00\n",
      "# reading input file \"/home/ubuntu/git-presentations/shark_mito/Carcharhinus_obscurus_mitochondrion.fasta\" of length 16706\n",
      "# matching query-file \"/home/ubuntu/git-presentations/shark_mito/Carcharhinus_obscurus_mitochondrion.fasta\"\n",
      "# against subject-file \"cut_vs_carcharhinus.ntref\"\n",
      "# COMPLETETIME /usr/bin/mummer cut_vs_carcharhinus.ntref 0.01\n",
      "# SPACE /usr/bin/mummer cut_vs_carcharhinus.ntref 0.03\n",
      "4: FINISHING DATA\n",
      "1\t8309\t8239\t16706\t8309\t8468\t91.96\t16476\t16706\t50.43\t50.69\ttig00000010:1-16476\tKC470543.1\n",
      "8310\t16476\t1\t8238\t8167\t8238\t95.29\t16476\t16706\t49.57\t49.31\ttig00000010:1-16476\tKC470543.1\n"
     ]
    }
   ],
   "source": [
    "nucmer -p cut_vs_carcharhinus tig00000010_cut.fasta Carcharhinus_obscurus_mitochondrion.fasta\n",
    "show-coords -lrcTH cut_vs_carcharhinus.delta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">We rotate the genome using samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools faidx tig00000010_cut.fasta 'tig00000010:1-16476':8310-16476 'tig00000010:1-16476':1-8309 | grep -v '>' | tr -d '\\n' | sed '1s/^/>tig00000010_cut_rotated\\n/' | sed -e '$a\\'  > tig00000010_cut_rotated.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">The dotplot show tig00000010 has the standar starting point now\n",
    "<img src=\"files/cut_rotated_vs_obscurus.jpeg\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\"> We use <a href=\"http://mitofish.aori.u-tokyo.ac.jp/annotation/input.html\">MitoFish server</a> to anotate tig00000010. this gives us a <a href=\"files/tig00000010_cut_rotated_genes.fa\">fasta file</a> with genes. We then blast them agains nt. \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "blastn -db nt -query tig00000010_cut_rotated_genes.fa -out genes.blast -outfmt \"6 qseqid sseqid pident length evalue sscinames salltitles sallacc sstart send qstart qend\" -max_target_seqs 1 -num_threads 16 -culling_limit 1 -max_hsps 1 -evalue 0.001"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">We count how many genes blast assings to each shark species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     22 Carcharhinus falciformis\n",
      "      9 Carcharhinus obscurus\n",
      "      1 Sphyrna lewini\n",
      "      1 Prionace glauca\n",
      "      1 Mustelus widodoi\n",
      "      1 Carcharhinus macloti\n",
      "      1 Carcharhinus brevipinna\n",
      "      1 Carcharhinus amblyrhynchos\n"
     ]
    }
   ],
   "source": [
    "cut -f6 genes.blast | sort | uniq -c | sort -nr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue;font-size:14pt\">With this information we can conclude that the sample was Carcharhinus falciformis"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
